### ISSUE TITLE
[FEATURE] implement hover anchor on more glyphs

### ISSUE THREAD

**Issue by kylrth** [bokeh/bokeh#8981]
#### Software version info

```
Bokeh 1.0.4
Python 3.7.3
jupyter_client 5.2.4
Windows 8 (64-bit)
Google Chrome 74.0.3729.169 (64-bit)
```

#### Description of expected behavior and the observed behavior

By specifying the keyword argument `anchor='center_right'` to `bokeh.models.HoverTool`, I expect to see the tooltips pointing to the right edge of each bar. Instead, the tooltips still point to the center of each bar.

#### Complete, minimal, self-contained example code that reproduces the issue

```python
import bokeh
from bokeh.plotting import figure

bokeh.io.output_notebook()

source = bokeh.models.ColumnDataSource(dict(
    lengths=[1, 2, 3, 4, 5],
    pos=[1, 2, 3, 4, 5]
))

hover = bokeh.models.HoverTool(
    tooltips=[
        ('pos', '@pos'),
        ('length', '@lengths')
    ],
    anchor='center_right'
)

fig = figure()
fig.hbar(y='pos', right='lengths', height=0.7, source=source)
fig.add_tools(hover)

bokeh.io.show(fig)

# bokeh.io.curdoc().add_root(fig)  # this also happens on the Bokeh server
```

#### Browser JavaScript console output when running `bokeh serve`

```
bokeh-1.0.4.min.js:31 [bokeh] setting log level to: 'info'
bokeh-1.0.4.min.js:31 [Violation] Added non-passive event listener to a scroll-blocking 'wheel' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
t @ bokeh-1.0.4.min.js:31
t.initialize @ bokeh-1.0.4.min.js:31
t @ bokeh-1.0.4.min.js:31
t @ bokeh-1.0.4.min.js:31
t @ bokeh-1.0.4.min.js:31
i.build_views @ bokeh-1.0.4.min.js:31
t.build_child_views @ bokeh-1.0.4.min.js:31
t.initialize @ bokeh-1.0.4.min.js:31
t @ bokeh-1.0.4.min.js:31
t @ bokeh-1.0.4.min.js:31
t @ bokeh-1.0.4.min.js:31
e @ bokeh-1.0.4.min.js:31
i @ bokeh-1.0.4.min.js:31
i.add_document_standalone @ bokeh-1.0.4.min.js:31
l @ bokeh-1.0.4.min.js:31
(anonymous) @ bokeh-1.0.4.min.js:31
bokeh-1.0.4.min.js:31 [bokeh] document idle at 69 ms
bokeh-1.0.4.min.js:31 [Violation] 'requestAnimationFrame' handler took 104ms
```

I would include a screenshot, but Windows 8's Snipping Tool doesn't have a delay feature so I can't mouse over to show the tooltip. I know.

**Comment by bryevdv** [comment]
So far, `anchor` has only been implemented on `Quad`.  For now, if you really need this, you will have to use `quad` instead of `hbar`. Adding to all the "box" glyphs will probably be simple (a good task fo a new contributor if you are interested) but should also update the `anchor` help string to make it clear that only certain  glyphs support it.

**Comment by kylrth** [comment]
I could possibly work on this next week.

**Comment by bryevdv** [comment]
@kylrth that would be awesome!

For reference, the relevant function on `quad.ts` is here:

https://github.com/bokeh/bokeh/blob/master/bokehjs/src/lib/models/glyphs/quad.ts#L25-L43

I think if you just literally move that function to `box.ts`:

https://github.com/bokeh/bokeh/blob/master/bokehjs/src/lib/models/glyphs/box.ts

Then things might automatically work for all "box" glyphs (quad, rect, vbar, hbar)

**Comment by bryevdv** [comment]
This was closed in #9013

1.
